packages/fuels-signers/src/wallet.rs

```diff
@@ -336,6 +420,98 @@ impl WalletUnlocked {
         Ok(Self::new_from_private_key(secret_key, provider))
     }
 
+    // Add base asset inputs to the transaction to cover the estimated fee.
+    // The original base asset amount cannot be calculated reliably from
+    // the existing transaction inputs because the selected coins may exceed
+    // the required amount to avoid dust. Therefore we require it as an argument.
+    //
+    // Requires contract inputs to be at the start of the transactions inputs vec
+    // so that their indexes are retained
+    pub async fn add_fee_coins(
+        &self,
+        tx: &mut Transaction,
+        previous_base_amount: u64,
+        witness_index: u8,
```

Comment:
Just some thoughts on usability here. 

From a UX perspective, I dislike having to leak `witness_index` at this level. For most users that just want to `add_fee_coins()` to their transactions, I believe this will sound very confusing. I'm seeing lots of `add_fee_coins(&mut tx, 0, 1)`, which is very unreadable :(.

I understand it's hard _not_ to need `previous_base_amount`. But how about not needing `witness_index` here? Possible? How hard? Would it be worth the work to improve the interface here? How often will this function be directly used by the users?

Is the decision between `add_fee_coins(&mut tx, amount, 0)` and `add_fee_coins(&mut tx, amount, 1)` only when it comes to deployment vs. regular calls? If so, would it be better to have one specific for deployment (e.g. `add_deployment_fee_coins()`) and one "regular" one (e.g. `add_fee_coins()` as-is)? 

Restricting the third param, which is pretty open right now (could be `0`, `1`, or any typo possible), by removing the third param and turning this decision into a simple English-based decision between two methods should be a better experience.

Thoughts?

Reply:
> How often will this function be directly used by the users?
Not often I assumed, since users rarely hand-craft their own transactions. And if they do, do they still need this?

> Restricting the third param, which is pretty open right now (could be 0, 1, or any typo possible), by removing the third param and turning this decision into a simple English-based decision between two methods should be a better experience.

I agree with this. However, I don't know all the possible scenarios for witness index. In our current code it is either 1 when deploying or 0 in all other cases. What are the other possibilities and are they worth considering?
I guess it's fine if we keep it as more of an internal helper.

Reply:
> Not often I assumed, since users rarely hand-craft their own transactions.

Ah, this is a key point. For a sec, I thought this would be required in simple contract calls (like most examples in our main test suite), but this isn't the case. So, for the majority of the use cases, it's still simple, and no need to know about `add_fee_coins()`.

If they're hand-crafting txs, most likely than not they'll know and think about witness index stuff, so `add_fee_coins()` will be at the same abstraction level, which is all good 😃.

I believe we can keep as-is, just add the comments, and also turn this method's top-level comment into a proper docstring (`\\\`), please!